scripts/RelayHubGasEstimation.ts

```diff
@@ -505,122 +567,101 @@ const estimateGas = async () => {
-      estimatedDestinationCallGas.toNumber()
-    );
+  logTitle('Gas estimation tests for token transfer - without token payment');
+  await estimateTokenTransferGasOverhead('0');
+}
 
-    const estimatedTokenPaymentGas = await ethers.provider.estimateGas({
-      from: forwarder.address,
-      to: token.address,
-      data: token.interface.encodeFunctionData('transfer', [
-        relayWorker.address,
-        fees,
-      ]),
-    });
+async function runDeployEstimation(tokenAmount = '0') {
+  const {owner, relayHub,factory, token, verifier} = await deployAndSetup();
 
-    const tokenGas = !isSponsored
-      ? {
-          tokenGas: correctEstimatedCallCost(
-            estimatedTokenPaymentGas.toNumber()
-          ),
-        }
-      : {};
-
-    const completeReq: RelayRequest = cloneRelayRequest(relayRequest, {
-      request: {
-        data: encodedFunction,
-        to: token.address,
-        nonce: (await forwarder.nonce()).toString(),
-        tokenAmount: fees,
-        tokenContract,
-        gas: internalDestinationCallCost,
-        ...tokenGas,
-      },
-    });
+  const GAS_PRICE = '60000000';
 
-    return completeReq;
+  if(tokenAmount !== '0'){
+    const swAddress = await factory.getSmartWalletAddress(owner.address, constants.AddressZero, '1');
+    await token.mint(tokenAmount, swAddress);
   }
 
-  async function estimateTokenTransferGasOverhead(fees: string) {
-    await deployAndSetup();
-    // refill SW balance
-    await token.mint('9000', forwarder.address);
+  const deployRequest = {
+    request: {
+      data: '0x00',
+      from: owner.address,
+      nonce: '1',
+      relayHub: relayHub.address,
+      to: constants.AddressZero,
+      tokenAmount,
+      tokenContract: constants.AddressZero,
+      tokenGas: '31259',
+      value: '0',
+      validUntilTime: 0,
+      index: 1,
+      recoverer: constants.AddressZero,
+    },
+    relayData: {
+      callForwarder: factory.address,
+      callVerifier: verifier.address,
+      feesReceiver: relayWorker.address,
+      gasPrice: GAS_PRICE,
+    },
+  } as DeployRequest;
 
-    const { forwarderInitialBalance, relayWorkerInitialBalance } =
-      await getCurrentBalances();
+  const { signature } = await getSuffixDataAndSignature(
+    factory,
+    deployRequest,
+    owner
+  );
 
-    const transferReceiver = ethers.Wallet.createRandom();
+  const txResponse = await relayHub
+    .connect(relayWorker)
+    .deployCall(deployRequest, signature, { gasPrice: GAS_PRICE });
 
-    // necessary to execute the transfer tx without relay
-    const balanceToTransfer = 1000;
+  const { gasUsed } = await txResponse.wait();;
 
-    // forge the request
-    const completeReq: RelayRequest = await forgeRequest(
-      transferReceiver.address,
-      balanceToTransfer,
-      fees,
-      token, relayRequest,
-      forwarder
-    );
+  console.log('Total gas used on deploy: ', gasUsed.toString());
```

Comment:
We're printing the total gas used for the deployment without highlighting the information (I'd use different colors compared to the ones used for the transfer overhead)

Reply:
I'm not really fan of using colors in log. I simplified the output of the script without colors. Have a look at it and give your opinion.